### PR TITLE
Conversation API: add integration tests

### PR DESCRIPTION
**Pull request by yaron2**
This PR adds integration tests to the conversational API, registers the echo component and fixes a few small serialization issues.

### CODE REVIEW

File: tests/integration/suite/daprd/conversation/grpc/basic.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	rtv1 "github.com/dapr/dapr/pkg/proto/runtime/v1"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	procdaprd "github.com/dapr/dapr/tests/integration/framework/process/daprd"
```
Comment (JoshVanL) [inline tests/integration/suite/daprd/conversation/grpc/basic.go:25]: nit
```suggestion
	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
```

File: tests/integration/suite/daprd/conversation/grpc/basic.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(basic))
+}
+
+type basic struct {
+	daprd *procdaprd.Daprd
```
Comment (JoshVanL) [inline tests/integration/suite/daprd/conversation/grpc/basic.go:34]: nit

```suggestion
	daprd *daprd.Daprd
```

File: tests/integration/suite/daprd/conversation/grpc/basic.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+	client := b.daprd.GRPCClient(t, ctx)
+
+	t.Run("good input", func(t *testing.T) {
+		resp, err := client.ConverseAlpha1(ctx, &rtv1.ConversationAlpha1Request{
+			Name:   "echo",
+			Inputs: []string{"well hello there"},
+		})
+		require.NoError(t, err)
+		require.Equal(t, "well hello there", resp.GetOutputs()[0].GetResult())
```
Comment (JoshVanL) [inline tests/integration/suite/daprd/conversation/grpc/basic.go:67]:
```suggestion
		require.Len(t, "well hello there", resp.GetOutputs(), 1)
		require.Equal(t, "well hello there", resp.GetOutputs()[0].GetResult())
```

File: tests/integration/suite/daprd/conversation/http/basic.go [hunk 4]
```diff
@@ -43,15 +43,14 @@ func (b *basic) Setup(t *testing.T) []framework.Option {
 apiVersion: dapr.io/v1alpha1
 kind: Component
 metadata:
-  name: conversation
+  name: echo
```
Comment (JoshVanL) [inline tests/integration/suite/daprd/conversation/http/basic.go:46]: nit: remove alias import of `procdaprd`.

File: tests/integration/suite/daprd/conversation/http/basic.go [hunk 5]
```diff
@@ -61,23 +60,32 @@ spec:
 func (b *basic) Run(t *testing.T, ctx context.Context) {
 	b.daprd.WaitUntilRunning(t, ctx)
 
-	postURL := fmt.Sprintf("http://localhost:%d/v1.0/conversation/echo", b.daprd.HTTPPort())
+	postURL := fmt.Sprintf("http://localhost:%d/v1.0-alpha1/conversation/echo/converse", b.daprd.HTTPPort())
```
Comment (JoshVanL) [inline tests/integration/suite/daprd/conversation/http/basic.go:63]:
```suggestion
	postURL := fmt.Sprintf("http://%sv1.0-alpha1/conversation/echo/converse", b.daprd.HTTPAddress())
```

**Review by JoshVanL — CHANGES_REQUESTED** [review]
Just some nits from me

File: tests/integration/suite/daprd/conversation/http/basic.go (hunk 4 above)
Comment (yaron2) [reply tests/integration/suite/daprd/conversation/http/basic.go:46]: Done

**Comment by yaron2** [comment]
> Just some nits from me

Addressed

**Review by JoshVanL — APPROVED** [review]

**Pull request merged**